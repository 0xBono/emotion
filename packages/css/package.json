{
  "name": "@emotion/css",
  "version": "0.9.8",
  "description": "a function to serialize css and object styless",
  "main": "dist/css.cjs.js",
  "module": "dist/css.esm.js",
  "license": "MIT",
  "repository": "https://github.com/emotion-js/emotion/tree/master/next-packages/css",
  "dependencies": {
    "@emotion/serialize": "^0.9.1",
    "@emotion/utils": "^0.8.2",
    "babel-plugin-emotion": "^9.2.10"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "src",
    "dist",
    "macro.js"
  ],
  "umd:main": "./dist/index.min.js",
  "browser": {
    "./dist/css.cjs.js": "./dist/css.browser.cjs.js",
    "./dist/css.esm.js": "./dist/css.browser.esm.js"
  }
}
