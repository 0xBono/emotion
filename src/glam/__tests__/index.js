/* eslint-disable jsx-quotes,no-useless-escape */
/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import plugin from '../babel'
import css, {fragment} from 'glam'

const babel = require('babel-core')

describe('emotion/glam', () => {
  describe('babel', () => {
    test('basic', () => {
      const basic = '(<div className="a" css={`color: brown;`}></div>)'
      const {code} = babel.transform(basic, {plugins: [plugin]})
      expect(code).toMatchSnapshot()
    })

    test('no css attr', () => {
      const basic = '(<div></div>)'
      const {code} = babel.transform(basic, {plugins: [plugin]})
      expect(code).toMatchSnapshot()
    })

    test('css empty', () => {
      const basic = '(<div css=""></div>)'
      const {code} = babel.transform(basic, {plugins: [plugin]})
      expect(code).toMatchSnapshot()
    })

    test('wrong value type', () => {
      const basic = '(<div css={5}></div>)'
      expect(() => babel.transform(basic, {plugins: [plugin]})).toThrow()
    })

    test('StringLiteral css prop value', () => {
      const basic = `<div css="color: brown;"></div>`
      const {code} = babel.transform(basic, {plugins: [plugin]})
      expect(code).toMatchSnapshot()
    })

    test('noClassName', () => {
      const basic = '(<div css={`color: brown;`}></div>)'
      const {code} = babel.transform(basic, {plugins: [plugin]})
      expect(code).toMatchSnapshot()
    })

    test('emptyClassName', () => {
      const basic = '(<div className="" css={`color: brown;`}></div>)'
      const {code} = babel.transform(basic, {plugins: [plugin]})
      expect(code).toMatchSnapshot()
    })

    test('className as expression', () => {
      const basic = '(<div className={variable} css={`color: brown;`}></div>)'
      const {code} = babel.transform(basic, {plugins: [plugin]})
      expect(code).toMatchSnapshot()
    })

    test('className as expression string', () => {
      const basic =
        '(<div className={`test__class\`} css={`color: brown;`} this={`hello\`}></div>)'
      const {code} = babel.transform(basic, {plugins: [plugin]})
      expect(code).toMatchSnapshot()
    })
  })

  describe('real', () => {
    test('basic', () => {
      const tree = renderer
        .create(
          <p css={`color: red;`}>
            hello world
          </p>
        )
        .toJSON()

      expect(tree).toMatchSnapshot()
    })

    // test('string expression', () => {
    //   const tree = renderer
    //     .create(
    //       <p css={'color: red;'}>
    //         hello world
    //       </p>
    //     )
    //     .toJSON()
    //
    //   expect(tree).toMatchSnapshot()
    // })

    test('kitchen sink', () => {
      const props = {online: false, error: false, radius: 5}
      const huge = 100
      const tiny = 6

      const bold = fragment`
        display: flex;
        font-weight: bold;`

      const big = fragment`
        @apply ${bold};
        font-size: ${huge}`

      const small = fragment`
        font-size: ${tiny}`

      const flexCenter = fragment`
        display: flex;
        justify-content: center;
        align-items: center`

      const tree = renderer
        .create(
          <div
            className="css__legacy-stuff"
            css={`
              @apply ${bold}
              @apply ${flexCenter};
             `}
          >
            <h1
              css={`
                @apply ${props.error ? big : small};
                color: red
              `}
            >
              BOOM
            </h1>
            <p className="test_class1" css={`color: blue;`}>Hello</p>
            <p
              className="test_class1 test___class45"
              css={`display: inline-flex`}
            >
              World
            </p>
            <p
              css={`
                color: red;
                border-radius: ${props.radius};
                &:hover {
                  font-weight: bold;
                  color: ${props.online ? 'green' : 'gray'};
                }
              `}
            >
              hello world
            </p>

          </div>
        )
        .toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
