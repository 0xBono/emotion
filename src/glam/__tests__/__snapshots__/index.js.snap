// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emotion/glam babel basic 1`] = `"<div className={\\"a\\" + \\" \\" + css\`color: brown;\`}></div>;"`;

exports[`emotion/glam babel className as expression 1`] = `"<div className={varia + \\" \\" + css\`color: brown;\`}></div>;"`;

exports[`emotion/glam babel className as expression string 1`] = `"<div className={\`test__class\` + \\" \\" + css\`color: brown;\`} this={\`hello\`}></div>;"`;

exports[`emotion/glam babel emptyClassName 1`] = `"<div className={\\"\\" + \\" \\" + css\`color: brown;\`}></div>;"`;

exports[`emotion/glam babel no css attr 1`] = `"<div></div>;"`;

exports[`emotion/glam babel noClassName 1`] = `"<div className={css\`color: brown;\`}></div>;"`;

exports[`emotion/glam real basic 1`] = `
<p
  className="css-jk0pkr"
>
  hello world
</p>
`;

exports[`emotion/glam real nested 1`] = `
<div
  className="a css-jf1v9l"
>
  <p
    className="foo css-14jcta0"
  >
    Hello
  </p>
  <p
    className="foo css-1fpqmhj"
  >
    World
  </p>
  <p
    className=" css-7jbqmo vars-1mtfi0j"
  >
    hello world
  </p>
</div>
`;
